Release Jobs - Jobs Service (#624)

HayenNico opened the pull request:
Refactoring jobs.service.ts for the jobs release. WIP

## Description / Motivation

The jobs service requires a refactor based on the new jobs schema and configuration.

## Fixes:

* #624 

## Tests included/Docs Updated?

- [ ] Included for each change/fix?
- [ ] Passing? (Merge will not be approved unless this is checked) 
- [ ] Docs updated?
- [ ] New packages used/requires npm install? 
- [ ] Toggle added for new features?

HayenNico committed:
Initial refactor for job creation and status updates

Updated service functions for job creation, status updates and querying.

HayenNico committed:
Change sendJobStartEmail

Suggestion for the change to the email at job start.
Old version was very specific to archive/retrieve workflows, which should now go to separate microservices instead.

HayenNico committed:
Minor refactor job.schema

HayenNico commented:
On the automatic emails from SciCat on job creation/update: Since archival/retrieval will be relegated to separate (micro-)services, do we even need these at all from SciCat or do we leave that to the other service only?

HayenNico committed:
Remove custom email code from jobs.service

HayenNico commented:
Email notification code is for now excluded from the jobs.service file. Could be reimplemented with a call to MailService directly from the controller in the future.

nitrosx commented:
Sending email should be executed by an external service, as SciCat does not have any knowledge of the IT infrastructure where has been installed.
IMHO, we should have an action to instruct SciCat to talk to an external service that is dedicated to send emails.

HayenNico committed:
Fix return types from jobs.service

Return types from jobs.service actions now match equivalent operations for datasets.service
